exercises/exercise4/checkSpecialCases.xquery

```diff
@@ -0,0 +1,149 @@
+xquery version "1.0";
+
+declare namespace local = "local";
+declare variable $gameInstance := fn:doc("GameInstance.xml")/mancalaGame;
+declare variable $currentPlayer := local:getCurrentPlayer();
+declare variable $oppositePit := local:getOppositePit($lastPit);
+
+(: This is the main function which triggers all other helper functions :)
+declare function local:checkSpecialCases($lastPit as xs:string) {
+    (: Check if last pit is from current player AND last pit was empty before AND opposite pit has any seeds? :)
+    if (local:getIsPitOfCurrentPlayer($lastPit, $currentPlayer) and local:getWasLastPitEmpty($lastPit) and local:getOppositePitHasSeeds($oppositePit)) then (
+        (: All checks have passed, move last pit and opposite pit seeds to current players store :)
+        local:fillPlayersStore(local:getCurrentPlayersStore($currentPlayer)) and local:setPitsToZero($lastPit)
+    )
+    else (
+        (: Do nothing - not all checks have passed :)
+    )
+};
+
+(: ---- Getter functions ---- :)
+(: Determine current player :)
+declare function local:getCurrentPlayer() { 
+    if ($gameInstance/turnOfPlayerOne/text() = "1") then (
+        let $result := "player1"
+        return $result
+    )
+    else (
+        let $result := "player2"
+        return $result
+    )
+};
+
+(: Determine, which store is currently relevant :)
+declare function local:getCurrentPlayersStore($currentPlayer as xs:string) {
+    if ($currentPlayer = "player1") then (
+        let $result := "store1"
+        return $result
+    )
+    else (
+        let $result := "store2"
+        return $result
+    )
+};
+
+(: Determine store seed count of current player :)
+declare function local:getStoreSeedCount($currentPlayer as xs:string) {
+    if ($currentPlayer = "player1") then (
+        let $result := $gameInstance/$currentPlayer/store1
+        return $result
+    )
+    else (
+        let $result := $gameInstance/$currentPlayer/store2
+        return $result
+    )
+};
+
+(: Check, if last pit is from the current player :)
+declare function local:getIsPitOfCurrentPlayer($lastPit as xs:string, $currentPlayer as xs:string) {
```

Comment:
The method should be renamed to "getIsHouseOfCurrentPlayer" and the else block should not be empty. Rather return false in the else block (else means, it's a store).